File: FindAllDisappearedInArray.java
//The algorithm finds the disappeared numbers in the range [1, n] in the array nums by marking the presence of each number using the sign of the corresponding index. 
//It then iterates through the modified array to identify the indices with positive values, adding 1 to obtain the disappeared numbers. 
//The time complexity is O(n), where n is the length of the input array nums, and the space complexity is O(1).
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        List<Integer> res = new ArrayList<>();
        for(int i=0;i<n;i++){
            int idx = Math.abs(nums[i]) -1;
            if(nums[idx]>0) nums[idx] *= -1;
        }
        for(int i=0;i<n;i++){
            if(nums[i]>0) res.add(i+1);
            else nums[i] *= -1;
        }return res;
        
    }
}

File: FindAllDisappearedInArray.py
# Time complexity - O(n) n is the number of elements
# Space complexity - O(1) we do not use any space for computation

class Solution:
    def findDisappearedNumbers(self, nums: list[int]) -> list[int]:
        # In this solution I am gonna iterate over list
        # Whatever value I get from a particular index, I just go that particular index and make value negative
        # At the end I will just check which indexes left positive and get those indexes + 1
        for i in nums:
            if nums[abs(i) - 1] > 0:
                nums[abs(i) - 1] *= -1
        result = []
        for i in range(len(nums)):
            if nums[i] > 0:
                result.append(i+1)
        return result
